src/offset_of.rs

```diff
@@ -66,30 +66,30 @@
 /// }
 /// ```
 #[macro_export]
-#[cfg(memoffset_constant_expression)]
+#[cfg(memoffset_maybe_uninit)]
 macro_rules! offset_of {
-    ($parent:ty, $($field:tt)+) => (unsafe {
-        let x: &'static $parent = $crate::Transmuter::<$parent> { int: 0 }.ptr;
-        $crate::Transmuter { ptr: &x.$($field)+ }.int
+    ($parent:ty, $field:tt) => (unsafe {
+        // Create an instance of the container and calculate the offset to its
+        // field. Although we are creating references to uninitialized data this
+        // is fine since we are not dereferencing them.
```

Comment:
This comment is not accurate. It's not fine. It's just the best we can do, currently.

The comment should IMO clearly say "yes, this is UB" and then explain why it does this anyway.
(The UB is in the `let &$container ...` line.)